anderson/CleanCodeExercise.js

```diff
@@ -0,0 +1,70 @@
+// “Always code as if the guy who ends up maintaining your code will be a violent psychopath who knows where you live”
+// 
+// THIS IS A JAVASCRIPT-LIKE CODE IMPLEMENTING THE SCHEDULING OF BATHS AND VETERINARIANS ON THE PET STORE CASE.
+// 
+// PLEASE REVIEW THIS CODE TO GET IT AS CLEAN AS POSSIBLE. JUSTIFY YOUR CHANGES.
+// 
+
+
+var c;
+
+public class Costumer {
+
+	constructor(name, address, petName1, petName2, petName3, petName4, phoneNumber) {
+		this.name = height;
+		this.address = address;
+		this.petNames = [petName1, petName2, petName3, petName4];
+		this.phoneNumber = phoneNumber;
+	}
+	public getCustomer(id) {
+		return $http.get('myServer.com/getCustomer?id=' + id);
+	}
+	static all() {
+		return $http.get('myServer.com/returnAllCustomers');
+	}
+};
+
+public class Schedule {
```

Comment:
Situação parecida com a da classe Customer. Podemos fazer a classe Schedule ser na verdade a mãe dos diversos tipos de Schedule, algo como:
`
    class Schedule {
        public static final String TYPE_BATH = "BANHO";
        public static final String TYPE_VET = "VETERINARIO";

        constructor(Date dateTime, int duracao, Customer customer, String description, String otherInfo, String type) {
            this.dia = dateTime.getDay();
            this.hora = dateTime.getHour();
            this.customer = customer;
            this.description = description;
            this.otherInfo = otherInfo;
            this.type = type;
        }

        /** WARNING: don't change this method because it's been used in another places.
         @deprecated
        **/
        public saveSchedule() {
            //return $http.post('myServer.com/saveSchedule', this);
            return scheduleService.create(this); //o scheduleService sabe em qual url realizar o post.
        }
    }

    class ScheduleBath extends Schedule {

        constructor(Date dateTime, int duracao, Customer customer, String description, String otherInfo) {
            parent::constructor(dateTime, duracao, customer, description, otherInfo, parent::TYPE_BATH);
        }
    }

    class ScheduleVeterinarian extends Schedule {

        constructor(Date dateTime, int duracao, Customer customer, String description, String otherInfo) {
            parent::constructor(dateTime, duracao, customer, description, otherInfo, parent::TYPE_VETERINARIAN);
        }
    }

/**
Até aqui fizemos uma especialização da Schedule em duas outras classes e mantivemos o método saveSchedule na classe Schedule por questões de compatibilidade com código legado, porém anotamos ela com @deprecated para que não seja utilizado daqui pra frente. O conteúdo dela também foi alterado de forma a utilizar a ScheduleService, dessa forma futuras novas alterações no comportamento do create dos agendamentos, não precisarão ser replicados nesse método legado também, ela já vai ser beneficiada por tais novas implementações.
**/

`